### ISSUE TITLE
Getting last block fails

### ISSUE THREAD

**Issue by silentlight** [vechain/connex#100]
Hi all,

When using:
```
const lastBlock = await this.connex.thor.block().get();
```

ocassionaly I get genesis block (height = 0).

Do you know what might be wrong.

**Comment by ehamery** [comment]
Have you consider including that check in the `block()` function?
`this.connex.thor.status.head.number` should never be `null` or `undefined...

**Comment by qianbin** [comment]
> Have you considered including that check in the `block()` function?
> `this.connex.thor.status.head.number` should never be `null` or `undefined`...
> `await this.connex.thor.block().get()` sometimes returns null. That is the same issue, isn't it?

`await this.connex.thor.block().get()` returns null only if the block specified by the given arg `revision` does not exists.

**Comment by ehamery** [comment]
If you do not provide any argument to the `block()` function, `await this.connex.thor.block().get()` should return the head block, so it should never be `null` and it sometimes is. That looks like a bug to me.
Same as `this.connex.thor.status.head.number` being `null`.

**Comment by qianbin** [comment]
> If you do not provide any argument to the `block()` function, `await this.connex.thor.block().get()` should return the head block, so it should never be `null` and it sometimes is. That looks like a bug to me.
> Same as `this.connex.thor.status.head.number` being `null`.

You're right. I'll look into the code. Can you provides some context of your code piece?

**Comment by ehamery** [comment]
I am basically scanning blocks and once in a while `await this.connex.thor.block().get()` returns `null`.
